### ISSUE TITLE
ring.util.codec/FormEncodeable impl for nil

### ISSUE THREAD

**Issue by tomoharu-fujita** [ring-clojure/ring-codec#3]
form-encode fails to encode if nil value is in map:

user=> (require '[ring.util.codec :as codec])
user=> (codec/form-encode (codec/form-decode "a&b=1"))
IllegalArgumentException No implementation of method: :form-encode\* of protocol: #'ring.util.codec/FormEncodeable found for class: nil  clojure.core/-cache-protocol-fn (core_deftype.clj:541)

It would be nicer if nil value treated as is.

user=> (extend-type nil
  #_=> codec/FormEncodeable
  #_=> (form-encode\* [_ _] ""))
nil
user=> (codec/form-encode (codec/form-decode "a&b=1"))
"b=1&a="

**Comment by buob** [comment]
This might be the wrong place to post this, but I'm getting this error when I try to set a cookie in the response, which doesn't totally make sense to me.

```clj
        {:status 302
         :headers {"Location" "/"}
         :cookies {"user_id" {:value (:id user)}
                   "auth_token" {:value (:auth-token user)}}}
```

**Comment by weavejester** [comment]
@buob: The exception is likely is likely caused by `(:id user)` or `(:auth-token user)` being `nil`.

**Comment by buob** [comment]
@weavejester 

1. thank you for the kind response to the n00b-ish question.  After sitting on that for over 24 hours I chose to post it, then immediately realized that was probably the problem, logged it, and deleted the comment.
2. thank you for being so attentive (at 1am your time, no less)
3. thank you for your dedicated work on these projects that make it easy for someone new to clojure like me get up and running.

You're a gentleman and a scholar.

**Comment by weavejester** [comment]
Fixed by #18
